src/Microsoft.VisualBasic.Core/src/Microsoft/VisualBasic/ErrObject.vb

```diff
@@ -0,0 +1,329 @@
+Imports System.Security
+Imports System.Security.Permissions
+Imports System.Runtime.ConstrainedExecution
+
+Namespace Microsoft.VisualBasic
+    
+    Public NotInheritable Class ErrObject
+
+        ' Error object private values
+        Private m_curException As Exception
+        Private m_curErl As Integer
+        Private m_curNumber As Integer
+        Private m_curDescription As String
+        Private m_NumberIsSet As Boolean
+        Private m_ClearOnCapture As Boolean
+        Private m_DescriptionIsSet As Boolean
+
+        Friend Sub New()
+            Me.Clear() 'need to do this so the fields are set to Empty string, not Nothing
+        End Sub
+
+        '============================================================================
+        ' ErrObject functions.
+        '============================================================================
+        Public ReadOnly Property Erl() As Integer
+            Get
+                Return m_curErl
+            End Get
+        End Property
+
+        Public Property Number() As Integer
+            Get
+                If m_NumberIsSet Then
+                    Return m_curNumber
+                End If
+
+                If Not m_curException Is Nothing Then
+                    Me.Number = MapExceptionToNumber(m_curException)
+                    Return m_curNumber
+                Else
+                    'The default case.  NOTE:  falling into the default does not "Set" the property.
+                    'We only get here if the Err object was previously cleared.
+                    Return 0
+                End If
+            End Get
+
+            Set(ByVal Value As Integer)
+                m_curNumber = MapErrorNumber(Value)
+                m_NumberIsSet = True
+            End Set
+        End Property
+
+        ''' <summary>
+        ''' Determines what the correct error description should be.
+        ''' If we don't have an exception that we are responding to then
+        ''' we don't do anything to the message.
+        ''' If we do have an exception pending, we morph the description
+        ''' to match the corresponding VB error.
+        ''' We also special case HRESULT exceptions to map to a VB description
+        ''' if we have one.
+        ''' </summary>
+        ''' <param name="Msg"></param>
+        ''' <returns></returns>
+        ''' <remarks></remarks>
+        Private Function FilterDefaultMessage(ByVal Msg As String) As String
+            Dim NewMsg As String
+
+            'This is one of the default messages, 
+            If m_curException Is Nothing Then
+                'Leave message as is
+                Return Msg
+            End If
+
+            Dim tmpNumber As Integer = Me.Number
+
+            If Msg Is Nothing OrElse Msg.Length = 0 Then
+                Msg = GetResourceString("ID" & CStr(tmpNumber))
+            ElseIf System.String.CompareOrdinal("Exception from HRESULT: 0x", 0, Msg, 0, Math.Min(Msg.Length, 26)) = 0 Then
```

Comment:
We don't currently localize, but I assume this text being compared against won't ever be localized?

Reply:
Added a test to try to catch the case where the text is localized.

---
In reply to: [274900350](https://github.com/dotnet/corefx/pull/36808#discussion_r274900350) [](ancestors = 274900350)

Reply:
Note that I removed "from HRESULT" from a string recently